CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - `Debias` mechanism for classification, ranking and auc metrics. New parameter `is_debiased` to `calc_from_confusion_df`, `calc_per_user_from_confusion_df` methods of classification metrics, `calc_from_fitted`, `calc_per_user_from_fitted` methods of auc and rankning (`MAP`) metrics, `calc_from_merged`, `calc_per_user_from_merged` methods of ranking (`NDCG`, `MRR`) metrics. ([#152](https://github.com/MobileTeleSystems/RecTools/pull/152))
 - `nbformat >= 4.2.0` dependency to `[visuals]` extra ([#169](https://github.com/MobileTeleSystems/RecTools/pull/169))
+- Implement `fit_partial()` for `ImplicitALSWrapperModel` and `LightFMWrapperModel` ([#179](https://github.com/MobileTeleSystems/RecTools/pull/179))
```

Comment:
I'm very excited about the new feature. Thank you for the PR!

Let's discuss what we're trying to achieve with `fit_partial`.
In implicit ALS has `partial_fit_users` and `partial_fit_items` methods that allow to update weights only for specific parts of users/items and also support feeding new users and items to already fitted model.
LightFM model has different logic. It has `fit_partial` method. From docs: "Unlike fit, repeated calls to this method will cause training to resume from the current model state". This method does not support adding new users/items. And it is not meant to update only parts of embeddings. 

What exactly do we want to achieve with `fit_partial` in rectools? I would say that first step would be to follow LightFM logic. We are missing the ability to resume training, out `fit` always refits models from the beginning, just like LightFM `fit`.

As for `partial_fit_users` and `partial_fit_items` - these are very model specific ALS methods. I wouldn't call them `fit_partial`. I would move them to a separate issue and PR.

@chezou @feldlime what do you think?

Reply:
Good point.

My motivation for adding `fit_partial` is to enable incremental training. For example, weekly batches for full training and daily incremental training to adopt new and existing user transactions.

I don't seriously care about preserving old users/items. Rather, shortening training time would be important.

Actually, LightFM has a known workaround for adding a new user/item. https://github.com/lyst/lightfm/issues/347#issuecomment-707829342 
I included that workaround into the patch. ea33c05

Reply:
Sounds good. Let me add tests for re-training with the same dataset.

Reply:
Added the tests 117c5c7

Reply:
@blondered I think https://github.com/MobileTeleSystems/RecTools/pull/179/commits/117c5c7a1a177d7820b530843a16a6635df9ed87 shows how it works for the use case 2.

Please let me know if you'd like to add anything else.

Reply:
Thanks! Sorry for the delay. It's summer and it was a vacation time :)